doc/sflow/sflow_hld.md

```diff
@@ -259,6 +297,48 @@ The following figure shows the sFlow sample packet path flow:
 4. The hsflowd daemon's HsflowdRx() is waiting on the specific genetlink family name's multicast group id and receives the encapsulated sample. The HsflowdRx parses and extracts the encapsulated sflow attributes and injects the sample to the hsflowd packet thread using takeSample().
 5. The hsflowd packet thread accumulates sufficient samples and then constructs an sFlow UDP datagram and forwards to the configured sFlow collectors.
 
+### 6.3b **Mirror On Drop (MOD) notification path**
+The following figure shows the MOD notification path flow:
+```mermaid
+sequenceDiagram
+
+box Kernel drops
+participant kernel
+end
+
+box h/w drops
+participant ASIC
+participant Driver
+end
+
+box NOS application
+participant GENETLINK
+participant hsflowd
+participant sFlow_Collector
+end
+
+kernel ->> GENETLINK : Drop with discard reason to NET_DM
+ASIC ->> Driver : Drop Event (pkt and drop metadata)
+Driver ->> GENETLINK :  Drop with discard reason to NET_DM
+GENETLINK ->> hsflowd : genetlink multicast message to group 'events'
+hsflowd ->> sFlow_Collector : sampleType DISCARD
+
+```
+1. **Packets dropped by the ASIC to genetlink**
+        1. The ASIC sends MOD notification to the ASIC driver.
+        2. The ASIC driver encapsulates the sample in a genetlink buffer and adds the following netlink attributes to the sample: `IIFINDEX`, `ORIGSIZE`, `TRAP_NAME`, `TRAP_GROUP`. The genetlink buffer is sent via `genlmsg_multicast()` to family `NET_DM` on multicast group 'events'.

```

Comment:
Also should driver include "HW_DROPS" to identify the drops by ASIC.

Reply:
1) I don't see the Linux Kernel setting SW_DROPS.
https://github.com/search?q=NET_DM_ATTR_SW_DROPS+repo%3Atorvalds%2Flinux&type=code

2) mod_dm driver would fill NET_DM_ATTR_ORIGIN as NET_DM_ORIGIN_SW for kernel drops. 
This can be used to distinguish h/w and s/w drops without having to change TL SAI.